drivers/digital-io/max22200/iio_max22200.c

```diff
@@ -0,0 +1,1234 @@
+		.show = max22200_iio_read_ch_mode,
+		.store = max22200_iio_write_ch_mode,
+		.priv = MAX22200_IIO_CH_DRIVE
+	},
+	{
+		.name = "channel_drive_available",
+		.show = max22200_iio_read_ch_mode_available,
+		.priv = MAX22200_IIO_CH_DRIVE,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "channel_side",
+		.show = max22200_iio_read_ch_mode,
+		.store = max22200_iio_write_ch_mode,
+		.priv = MAX22200_IIO_CH_SIDE
+	},
+	{
+		.name = "channel_side_available",
+		.show = max22200_iio_read_ch_mode_available,
+		.priv = MAX22200_IIO_CH_SIDE,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "channel_op_mode",
+		.show = max22200_iio_read_ch_mode,
+		.store = max22200_iio_write_ch_mode,
+		.priv = MAX22200_IIO_CH_OP_MODE
+	},
+	{
+		.name = "channel_op_mode_available",
+		.show = max22200_iio_read_ch_mode_available,
+		.priv = MAX22200_IIO_CH_OP_MODE,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "channel_freq",
+		.show = max22200_iio_read_channel_freq,
+		.store = max22200_iio_write_channel_freq,
+	},
+	{
+		.name = "channel_freq_available",
+		.show = max22200_iio_read_channel_freq_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "slew_rate_control",
+		.show = max22200_iio_read_enable,
+		.store = max22200_iio_write_enable,
+		.priv = MAX22200_SRC
+	},
+	{
+		.name = "open_load_detection",
+		.show = max22200_iio_read_enable,
+		.store = max22200_iio_write_enable,
+		.priv = MAX22200_OL_ENABLE
+	},
+	{
+		.name = "dpm_enable",
+		.show = max22200_iio_read_enable,
+		.store = max22200_iio_write_enable,
+		.priv = MAX22200_DPM_ENABLE
+	},
+	{
+		.name = "hit_currrent_check",
+		.show = max22200_iio_read_enable,
+		.store = max22200_iio_write_enable,
+		.priv = MAX22200_HHF_ENABLE
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_attribute max22200_debug_attrs[] = {
+	{
+		.name = "thermal_protection_mask",
+		.show = max22200_iio_read_fault_mask,
+		.store = max22200_iio_write_fault_mask,
+		.priv = MAX22200_M_OVT
+	},
+	{
+		.name = "overcurrent_protection_mask",
+		.show = max22200_iio_read_fault_mask,
+		.store = max22200_iio_write_fault_mask,
+		.priv = MAX22200_M_OCP
+	},
+	{
+		.name = "open_load_mask",
+		.show = max22200_iio_read_fault_mask,
+		.store = max22200_iio_write_fault_mask,
+		.priv = MAX22200_M_OLF
+	},
+	{
+		.name = "hit_current_fault_mask",
+		.show = max22200_iio_read_fault_mask,
+		.store = max22200_iio_write_fault_mask,
+		.priv = MAX22200_M_HHF
+	},
+	{
+		.name = "dpm_fault_mask",
+		.show = max22200_iio_read_fault_mask,
+		.store = max22200_iio_write_fault_mask,
+		.priv = MAX22200_M_DPM
+	},
+	{
+		.name = "com_err_mask",
+		.show = max22200_iio_read_fault_mask,
+		.store = max22200_iio_write_fault_mask,
+		.priv = MAX22200_M_COMF
+	},
+	{
+		.name = "uvlo_mask",
+		.show = max22200_iio_read_fault_mask,
+		.store = max22200_iio_write_fault_mask,
+		.priv = MAX22200_M_UVM
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_attribute max22200_global_attrs[] = {
+	{
+		.name = "device_frequency",
```

Comment:
is this the sampling frequency of the device? let's try and use some generic attribute names if they apply. check this list: https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-bus-iio

Reply:
it's the chopping frequency of the device, can switch name to chopping_frequency if needed.

Reply:
yes, the name would be cleared

Reply:
Changing to chopping_frequency then.